math/src/main/kotlin/ktx/math/ImmutableVector2.kt

```diff
@@ -0,0 +1,182 @@
+
+    @Deprecated(
+            message = "use rotateDeg instead. (this function is not provided to be consistent with angleDeg)",
+            replaceWith = ReplaceWith("rotateDeg(angle)")
+    )
+    inline fun rotate(angle: Float): ImmutableVector2 = rotateDeg(angle)
+
+    companion object {
+
+        /** Vector zero */
+        val ZERO = ImmutableVector2(0f, 0f)
+
+        /** Unit vector of positive x axis */
+        val X = ImmutableVector2(1f, 0f)
+
+        /** Unit vector of positive y axis */
+        val Y = ImmutableVector2(0f, 1f)
+
+        /**
+         * Returns the [ImmutableVector2] represented by the specified string according to the format of [ImmutableVector2::toString]
+         */
+        fun fromString(string: String): ImmutableVector2 =
+                Vector2().fromString(string).toImmutableVector2()
+    }
+}
+
+/** @return an instance of [ImmutableVector2] with the same x and y values */
+inline fun ImmutableVector2.toVector2(): Vector2 = Vector2(x, y)
+
+/** @return an instance of [Vector2] with the same x and y values */
+inline fun Vector2.toImmutableVector2(): ImmutableVector2 = ImmutableVector2(x, y)
+
+/** Returns the angle in degrees of this vector relative to the [reference]. Angles are towards the positive y-axis (typically counter-clockwise.) between -180 and +180 */
+inline fun ImmutableVector2.angleDeg(reference: ImmutableVector2 = ImmutableVector2.X): Float =
+        angleDeg(reference.x, reference.y)
+
+/** Returns the angle in degrees of this vector relative to the reference vector described by [x] and [y]. Angles are towards the positive y-axis (typically counter-clockwise.) between -180 and +180 */
+inline fun ImmutableVector2.angleDeg(x: Float, y: Float): Float =
+        angleRad(x, y) * MathUtils.radiansToDegrees
+
+/** Returns a vector of same length rotated by the given [angle] in degree */
+inline fun ImmutableVector2.rotateDeg(angle: Float): ImmutableVector2 =
+        rotateRad(angle * MathUtils.degreesToRadians)
+
+/**
+ * Returns a vector of same length rotated by 90 degrees in the given [dir]
+ *
+ * @param dir positive value means toward positive y-axis (typically counter-clockwise). Negative value means toward negative y-axis (typically clockwise).
+ */
+inline fun ImmutableVector2.rotate90(dir: Int): ImmutableVector2 =
+        if (dir >= 0) copy(x = -y, y = x) else copy(x = y, y = -x)
+
+/** Returns a vector of same length with the given [angle] in degree */
+inline fun ImmutableVector2.withAngleDeg(angle: Float): ImmutableVector2 =
+        withAngleRad(angle * MathUtils.degreesToRadians)
+
+/** Calculates the 2D cross product between this and the [other] vector */
+inline infix fun ImmutableVector2.x(other: ImmutableVector2): Float = crs(other.x, other.y)
+
+/** Calculates the 2D cross product between this and the [other] vector */
+inline infix fun ImmutableVector2.crs(other: ImmutableVector2): Float = crs(other.x, other.y)
```

Comment:
Again, I'm not sure all of these should be inlined.

Reply:
They are only aliases.

Of course the policy of inlining aliases is debatable and I'll remove the inline for all functions if you want.

I just thought: the less method count the better. So if an function is only an *alias* (like here) I'd inline.

Just tell me if you want me to remove the aliases, and I'll do it.